### PR TITLE
Build fea files in ufo+ds if identical across masters

### PR DESCRIPTION
**Pull request by rsheeter**
Baby steps.

Final stage of #76 broken into smaller parts.

Fixes https://github.com/googlefonts/fontmake-rs/issues/73

Along the way fixed https://github.com/googlefonts/fontmake-rs/issues/45.

### CODE REVIEW

**Review by cmyr — APPROVED** [review]
exciting!

File: fontbe/src/features.rs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    /// Inspired by (as in shameless copy of) how the fea-rs binary flows.
+    fn compile(&self, fea_file: &Path, glyph_order: GlyphMap) -> Result<FontBuilder, Error> {
+        // Will you not parse?!
+        let parse =
+            fea_rs::parse_root_file(fea_file, Some(&glyph_order), Some(self.build_dir.clone()))
+                .map_err(|e| Error::FeaError(format!("{:?} parsing {:?}", e, fea_file)))?;
+        let (tree, diagnostics) = parse.generate_parse_tree();
+        check_diagnostics(fea_file, "generate parse tree", &diagnostics, |d| {
+            format!("{:?}", d)
+        })?;
```
[the remark this answers is not in the document]
Comment (rsheeter) [reply fontbe/src/features.rs:58]: https://github.com/cmyr/fea-rs/issues/104 :)

File: fontir/src/serde.rs [hunk 2]
```diff
@@ -63,6 +63,48 @@ impl From<StaticMetadata> for StaticMetadataSerdeRepr {
     }
 }
 
+// The HashMap<NormalizedLocation, GlyphInstance> seems to throw serde for a loop sometimes
```
Comment (anthrotype) [inline fontir/src/serde.rs:66]: I think this specific change deserved its own issue/PR instead of being placed inside this large and seemingly unrelated commit..

File: fontir/src/serde.rs [hunk 3]
```diff
@@ -63,6 +63,48 @@ impl From<StaticMetadata> for StaticMetadataSerdeRepr {
...
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct GlyphInstanceSerdeRepr {
+    location: NormalizedLocation,
+    instance: GlyphInstance,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct GlyphSerdeRepr {
+    pub name: String,
+    pub instances: Vec<GlyphInstanceSerdeRepr>,
```
Comment (anthrotype) [inline fontir/src/serde.rs:76]: we decided to call that attribute "sources" in the fontir::ir::Glyph struct which this GlyphSerdeRepr is meant to be associated with, I don't see why use "instances" instead of the same "sources" field name
